src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverter.cs

```diff
@@ -69,6 +71,13 @@ internal virtual object CreateObject(JsonSerializerOptions options)
 
         internal abstract JsonParameterInfo CreateJsonParameterInfo();
 
+        internal virtual JsonConverter<TargetType> CreateCastingConverter<TargetType>()
```

Comment:
I would just follow existing convention here: mark as abstract and throw in the factory implementation.